paasta_tools/cli/cmds/local_run.py

```diff
@@ -627,7 +631,9 @@ def run_docker_container(
             cluster_name=instance_config.cluster,
             secret_provider_kwargs=secret_provider_kwargs,
         )
-        environment.update(secret_environment)
+        for k in secret_environment.keys():
```

Comment:
This seems strange to me. before this line updated the environment, now we are deleting it. Why do we have to delete it if it wasn't in there in the first place?

Reply:
If you don't delete secret envs, you will see something like this:

--env MY_SECRET_VAR=SECRET(test-secret) ... --env MY_SECRET_VAR

The first part comes from environment and the second part comes from secret_environment.

Reply:
It's even worse isn't it because `SECRET(test-secret)` will have been decrypted at that point. So the one thing we want to hide is actually still being shown!

Reply:
No, "--env MY_SECRET_VAR=SECRET(test-secret)" won't be shown. Only "--env MY_SECRET_VAR" is shown. That is why I delete secret_environment from the super set environment dict.